### PR TITLE
[cleanup] Remove unused parameters in method

### PR DESCRIPTION
**Pull request by gaozhangmin**
### Motivation
There are so many unused parameters of method .

### Modifications

Remove unused parameters of method 

### Documentation

Check the box below or label this PR directly.

Need to update docs? 

- [ ] `doc-required` 
(Your PR needs to update docs and you will update later)
  
- [ ] `no-need-doc` 
(Please explain why)
  
- [x] `doc` 
(Your PR contains doc changes)

- [ ] `doc-added`
(Docs have been already added)

### CODE REVIEW

**Review by HQebupt — APPROVED** [review]

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/BrokerFilter.java [hunk 1]
```diff
@@ -36,15 +35,13 @@ public interface BrokerFilter {
...
-     *            The data for the bundle to assign.
      * @param loadData
      *            The load data from the leader broker.
      * @param conf
      *            The service configuration.
      * @throws BrokerFilterException
      *            There was an error in the pipeline and the brokers should be reset to their original value
      */
-    void filter(Set<String> brokers, BundleData bundleToAssign, LoadData loadData, ServiceConfiguration conf)
+    void filter(Set<String> brokers, LoadData loadData, ServiceConfiguration conf)
```
Comment (Jason918) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/BrokerFilter.java:45]: It's better not to change this. Some developer may use custom filter class. Let's keep the expansibility。

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/impl/SimpleResourceAllocationPolicies.java [hunk 2]
```diff
@@ -38,7 +34,7 @@ public SimpleResourceAllocationPolicies(PulsarService pulsar) {
         this.pulsar = pulsar;
     }
 
-    public boolean canAssign(ServiceUnit srvUnit, ResourceUnit rescrUnit, Map<ResourceUnit, LoadReport> loadReports) {
+    public boolean canAssign() {
```
Comment (Jason918) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/impl/SimpleResourceAllocationPolicies.java:37]: This method seems to be a partial work. Better not change it.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java [hunk 3]
```diff
@@ -2314,7 +2314,8 @@ private void updateConfigurationAndRegisterListeners() {
         }
 
         // add listener to notify topic subscriptionTypesEnabled changed.
-        registerConfigurationListener("subscriptionTypesEnabled", this::updateBrokerSubscriptionTypesEnabled);
+        registerConfigurationListener("subscriptionTypesEnabled",
```
Comment (Jason918) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java:2317]: Personally, I am in favor of the previous version which save one anonymous lambda.

File: pulsar-broker/src/main/java/org/apache/pulsar/common/naming/NamespaceBundleFactory.java [hunk 4]
```diff
@@ -77,7 +76,7 @@ public NamespaceBundleFactory(PulsarService pulsar, HashFunction hashFunc) {
 
         this.bundlesCache = Caffeine.newBuilder()
                 .recordStats()
-                .buildAsync(this::loadBundles);
+                .buildAsync((namespace, executor) -> loadBundles(namespace));
```
Comment (Jason918) [inline pulsar-broker/src/main/java/org/apache/pulsar/common/naming/NamespaceBundleFactory.java:79]: Same here.

**Review by Jason918 — APPROVED** [review]
Overall LGTM, left some minor comments.

**Review by Technoboy- — APPROVED** [review]
LGTM
@mattisonchao Please help review this patch.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java [hunk 5]
```diff
@@ -2792,7 +2790,8 @@ private TopicName validateTopicName(String topic, long requestId, Object request
     }
 
     public ByteBufPair newMessageAndIntercept(long consumerId, long ledgerId, long entryId, int partition,
-            int redeliveryCount, ByteBuf metadataAndPayload, long[] ackSet, String topic, long epoch) {
```
Comment (michaeljmarshall) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java:2795]: I don't believe we should remove this method for the same reasons as above.

**Review by michaeljmarshall — CHANGES_REQUESTED** [review]
@gaozhangmin - thank you for your contribution. Because Pulsar has several extension points, I think we should be conservative when updating existing public methods that could reasonably be called by third part classes. I think the right step is to add a new method without the unused variable, and then deprecate the existing method.

**Comment by michaeljmarshall** [comment]
@Technoboy- what is the justification for cherry-picking this to release branches?

**Comment by Technoboy-** [comment]
> @Technoboy- what is the justification for cherry-picking this to release branches?

This patch has changed the signature of the methods. If not cherry-pick to other branch, later to these related fixes will fail.
@michaeljmarshall @gaozhangmin

**Comment by Technoboy-** [comment]
> > @Technoboy- what is the justification for cherry-picking this to release branches?
> 
> This patch has changed the signature of the methods. If not cherry-pick to other branch, later to these related fixes will fail. @michaeljmarshall @gaozhangmin

You are right @michaeljmarshall .

**Review by eolivelli — CHANGES_REQUESTED** [review]
I think that this kind of patches is dangerous and it also makes it harder to cherry-pick changes to other branches.

I suggest at least to split the patch in smaller patches, this way:
- it is easier to review
- it is less probable that we break things

Therefore, unfortunately in Pulsar we have many integration points that allow custom code (like Protocol Handlers) to access Pulsar internals. So changing the signatures without considering case per case may break the compatibility with existing protocol handlers.

I appreciate this effort but this is something that we should do carefully

**Pull request closed without merging**